Tutorial.md

```diff
@@ -0,0 +1,530 @@
+
+## Prerequisites<a name="resource-type-walkthrough-prereqs"></a>
+
+For purposes of this walkthrough, it is assumed you have already installed a node and npm versions as well as the [aws-cli](https://github.com/aws/aws-cli/tree/v2), the [cloudformation-cli](https://github.com/aws-cloudformation/cloudformation-cli) and the [Typescript plugin](https://github.com/eduardomourar/cloudformation-cli-typescript-plugin).
+
+## Create the Resource Provider Development Project<a name="resource-type-walkthrough-model"></a>
+
+Before we can actually design and implement our resource provider, we'll need to generate a new resource type project\.
+
+### Initiate the project<a name="resource-type-walkthrough-model-initiate"></a>
+
+1. Use the `init` command to create your resource provider project and generate the files it requires\.
+
+    ```
+    $ cfn init
+    Initializing new project
+    ```
+
+2. The `init` command launches a wizard that walks you through setting up the project, including specifying the resource name\. For this walkthrough, specify `Testing::Unicorn::Maker`\.
+
+    ```
+    Enter resource type identifier (Organization::Service::Resource)
+    >> Testing::Unicorn::Maker
+    ```
+
+   The wizard then enables you to select the appropriate language plugin\. Select Typescript\.
+
+    ```
+    Select a language for code generation:
+    [1] go
+    [2] java
+    [3] python36
+    [4] python37
+    [5] typescript
+    (enter an integer):
+    >> 5
+    ```
+
+3. Finally, you will be prompted for the use of Docker. Select Yes\.
+
+    ```
+    Use docker for platform-independent packaging (Y/n)?
+    This is highly recommended unless you are experienced
+    with cross-platform Typescript packaging.
+    >> Y
+    Initialized a new project in /Users/<user_id>/Projects/UnicornMaker
+    ```
+
+Intiating the project includes generating the files needed to develop the resource provider\. For example:
+
+```
+$ ls -1
+README.md
+docs
+example_inputs
+package.json
+resource-role.yaml
+rpdk.log
+sam-tests
+src
+template.yml
+testing-unicorn-maker.json
+tsconfig.json
+```
+
+## Setup Development Project<a name="resource-type-walkthrough-model-setup"></a>
+
+Update the `package.json` file by changing the description and dependencies to:
+
+```
+.
+.
+.
+"description": "Unicorn-maker is a complete example of a Cloudformation provider. This resource is built in multiple languages, to get you up and running creating Cloudformation custom resources.",
+.
+.
+.
+"dependencies": {
+    "cfn-rpdk": "https://github.com/eduardomourar/cloudformation-cli-typescript-plugin/releases/download/v0.3.3/cfn-rpdk-0.3.3.tgz",
+    "class-transformer": "^0.3.1",
+    "node-fetch": "^3.0.0-beta.4"
+},
+.
+.
+.
+```
+
+Run `npm install`
```

Comment:
this should be `npm install --optional` so that the AWS SDK, which is an optional dependency, gets installed